packages/techdocs-cli/src/commands/index.ts

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { CommanderStatic } from "commander";
+
+export function registerCommands(program: CommanderStatic) {
+  program
+    .command("generate")
+    .description("Generate TechDocs documentation site using mkdocs.")
+    .option(
+      "--source-dir <PATH>",
+      "Source directory containing mkdocs.yml and docs/ directory.",
+      "."
+    )
+    .option(
+      "--output-dir <PATH>",
+      "Output directory containing generated TechDocs site.",
+      "./site/"
+    )
+    .option(
+      "--no-docker",
+      "Do not use Docker, use MkDocs executable and plugins in current user environment."
+    )
+    .option(
+      "--techdocs-ref <HOST_TYPE:URL>",
+      "The repository hosting documentation source files e.g. github:https://ghe.mycompany.net.com/org/repo." +
+        "\nThis value is same as the backstage.io/techdocs-ref annotation of the corresponding Backstage entity." +
+        "\nIt is completely fine to skip this as it is only being used to set repo_url in mkdocs.yml if not found.\n"
+    )
+    .option("-v --verbose", "Enable verbose output.", false)
+    .alias("build")
+    .action(lazy(() => import("./generate/generate").then(m => m.default)));
+
+  program
+    .command("publish")
+    .description(
+      "Publish generated TechDocs site to an external storage AWS S3, Google GCS, etc."
+    )
+    .requiredOption(
+      "--publisher-type <TYPE>",
+      "(Required) awsS3 | googleGcs - same as techdocs.publisher.type in Backstage app-config.yaml"
+    )
+    .requiredOption(
+      "--bucket-name <BUCKET>",
```

Comment:
That's a great catch. Azure doesn't use 'bucket' as being pointed out in https://github.com/backstage/backstage/pull/4017

I imagine going forward, we will have a generic `--publisher-type` which accepts unique identifier strings - 'awsS3', 'googleGcs', 'azureStorage', etc.

And based on what value was given in a command, few options would become "required" and few would be unused.

Example

If --publisher-type is `awsS3` or `googleGcs` -> `--bucket-name` becomes required.
But if --publisher-type is `azureSomething` -> `--azure-<bucket-equivalent>-name` becomes required.

We need to indiciate these conditionals in the docs - what is required when. We already have conditional `app-config.yaml` rules based on publisher types. https://backstage.io/docs/features/techdocs/configuration

Reply:
I think renaming `--bucket-name` to `--storage-name` may cover AWS/GCS/Azure at least. GitHub Pages, etc. might be a totally different type of publisher.

Reply:
As I was adding some comments above, I ended up here after and yes, I agree with both of you. And for know, as you refer to it as "storage" in descriptions of the command, I think it makes sense to use that instead of bucket. Even though the "storage" is maybe much more than the actual bucket...?

Reply:
AWS and GCS folks will be happy with `--bucket-name`. Azure folks may be disappointed.

AWS/GCS/Azure/etc. may  neither be happy nor disappointed with `--storage-name`, as long as it says in the description that for AWS/GCS it means bucket name. That's my guess. 😄

Reply:
We can always add aliases in future, if requested by the community, since `--bucket-name` is more apt for some groups.